From 7a4fa794c72012bebe8cde3d33ccb925ff520933 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Wed, 8 Jun 2022 09:53:19 +0800
Subject: [PATCH 55/98] desktop-shell: Fix crash when destroying surface with
 null fade

Add null check for wsurface_anim_fade to prevent crash when destroying
surfaces that haven't committed animation states.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 desktop-shell/shell.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/desktop-shell/shell.c b/desktop-shell/shell.c
index f3944761..985dd920 100644
--- a/desktop-shell/shell.c
+++ b/desktop-shell/shell.c
@@ -156,7 +156,10 @@ desktop_shell_destroy_surface(struct shell_surface *shsurf)
 	weston_view_destroy(shsurf->view);
 
 	wl_signal_emit(&shsurf->destroy_signal, shsurf);
-	weston_surface_unref(shsurf->wsurface_anim_fade);
+
+	/* wsurface_anim_fade may be NULL if surface was never committed */
+	if (shsurf->wsurface_anim_fade)
+		weston_surface_unref(shsurf->wsurface_anim_fade);
 
 	if (shsurf->output_destroy_listener.notify) {
 		wl_list_remove(&shsurf->output_destroy_listener.link);
-- 
2.39.5

